agents/agents/executor/executor.go

```diff
@@ -706,21 +724,21 @@ func (e Executor) streamLogs(ctx context.Context, grpcClient pbscribe.ScribeServ
 //
 //nolint:cyclop,gocognit
 func (e Executor) processLog(parentCtx context.Context, log ethTypes.Log, chainID uint32) (err error) {
-	contractEvent := e.logType(log, chainID)
+	contractType := e.logType(log, chainID)
 
 	ctx, span := e.handler.Tracer().Start(parentCtx, "processLog", trace.WithAttributes(
 		attribute.Int(metrics.ChainID, int(chainID)),
-		attribute.Int("contract", int(contractEvent.contractType)),
-		attribute.Int("event", int(contractEvent.eventType)),
+		attribute.Int("contract", int(contractType)),
 		attribute.String(metrics.TxHash, log.TxHash.String()),
 	))
 
 	defer func() {
 		metrics.EndSpanWithErr(span, err)
 	}()
 
-	switch contractEvent.contractType {
-	case originContract:
+	//nolint:exhaustive
+	switch contractType {
+	case execTypes.OriginContract:
 		message, err := e.logToMessage(log, chainID)
```

Comment:
There's 0 reason to ever return a pointer to an interface in go. 

e.logToMessage can return types.Message. You can still check if types.Message is nil w/ `if messsage == nil return nil`

Than you can avoid all this dereference crap below. This goes for message, snapshot, and attestation.

Pointers should only be used on structs since those can't be nil. This is one reason we prefer interfaces in those type files in general